workers/modules/shared-worker-parse-error-failure.html

```diff
@@ -0,0 +1,45 @@
+<!DOCTYPE html>
+<title>SharedWorker: parse error failure</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+
+// Check if module shared worker is supported.
+// In this test scope, we only use simple non-nested static import as a feature
+// of module shared worker, so we only check if static import is supported.
+//
+// This check is necessary to appropriately test parse error handling because
+// we need to distingusih the parse error invoked by unsupported "import" in
+// the top-level script from the parse error invoked by the statically imported
+// script which is the one we want to check in this test.
+let supportsModuleWorkers = false;
+promise_test(async () => {
+  const worker = new SharedWorker('resources/static-import-worker.js',
+                                  { type: 'module' });
+  const msg = await new Promise((resolve, reject) => {
+    worker.port.onmessage = resolve;
+    worker.onerror = reject;
+  });
+  assert_array_equals(msg.data, ['export-on-load-script.js']);
+  if (msg.data.length == 1 && msg.data[0] === 'export-on-load-script.js') {
```

Comment:
If the assert fails then this line will not be reached, so this if condition is redundant.

I think the best way to do this is to convert this all into a `promise_setup()`.

```js
promise_setup(async () => {
  const worker = new SharedWorker('resources/static-import-worker.js',
                                  { type: 'module' });

  const supportsModuleWorkers = await new Promise((resolve, reject) => {
    worker.port.onmessage = e => resolve(e.data.length === 1 && e.data[0] == 'export-on-load-script.js');
    worker.onerror = () => resolve(false);
  });

  assert_precondition(supportsModuleWorkers, "Static import must be supported on module shared worker to run this test.");
});
```

Then you can remove the `assert_precondition` from each individual test. Also it makes it so that there are only 2 tests, both with precondition failures, instead of 1 test with a failure and 2 tests with precondition failures.

Reply:
Thanks. I modified in this way.